cmd/getcreds/main.go

```diff
@@ -115,84 +104,158 @@ func loadVerifyConfigFile(configFilename string) (AppConfigFile, error) {
+	//create attachment....
 	bodyBuf := &bytes.Buffer{}
 	bodyWriter := multipart.NewWriter(bodyBuf)
 
-	fileWriter, err := bodyWriter.CreateFormFile("pubkeyfile", pubKeyFilename)
+	//
+	fileWriter, err := bodyWriter.CreateFormFile("pubkeyfile", "somefilename.pub")
 	if err != nil {
+		fmt.Println("error writing to buffer")
+		//t.Fatal(err)
 		return nil, err
 	}
-	// open file handle
-	fh, err := os.Open(pubKeyFilename)
-	if err != nil {
-		return nil, err
-	}
-	defer fh.Close()
+	//when using a file this used to be: fh, err := os.Open(pubKeyFilename)
+	fh := strings.NewReader(filedata)
 
 	//iocopy
 	_, err = io.Copy(fileWriter, fh)
 	if err != nil {
-		log.Fatal(err)
+		//t.Fatal(err)
 		return nil, err
 	}
-	if *debug {
-		log.Printf("%v", bodyBuf)
-	}
 
 	contentType := bodyWriter.FormDataContentType()
 	bodyWriter.Close()
 
-	req, err := http.NewRequest("POST", targetUrl, bodyBuf)
+	// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
+	// pass 'nil' as the third parameter.
+	req, err := http.NewRequest(method, urlStr, bodyBuf)
 	if err != nil {
+		//t.Fatal(err)
 		return nil, err
 	}
 	req.Header.Set("Content-Type", contentType)
 
-	req.SetBasicAuth(userName, string(password[:]))
 	return req, nil
 }
 
-func getCertFromTargetUrls(pubKeyFilename, userName string, password []byte, targetUrls []string, rootCAs *x509.CertPool) (cert []byte, err error) {
-	success := false
-	for _, baseUrl := range targetUrls {
-		targetUrl := baseUrl + userName
-		log.Printf("attempting to target '%s'", targetUrl)
-		tlsConfig := &tls.Config{RootCAs: rootCAs, MinVersion: tls.VersionTLS12}
-		tr := &http.Transport{
-			TLSClientConfig: tlsConfig,
-		}
-		client := &http.Client{Transport: tr, Timeout: time.Duration(5) * time.Second}
+func doCertRequest(client *http.Client, authCookies []*http.Cookie, url, filedata string) ([]byte, error) {
 
-		// TODO: change basic auth for some form of tokens.
-		//req, err := http.NewRequest("POST", targetUrl, bodyBuf)
-		req, err := buildGetCredRequestBasicAuth(pubKeyFilename, userName, password, targetUrl)
-		if err != nil {
-			log.Fatal(err)
-		}
+	req, err := createKeyBodyRequest("POST", url, filedata)
+	if err != nil {
+		return nil, err
+	}
+	//add the login cookies
+	for _, cookie := range authCookies {
+		req.AddCookie(cookie)
+	}
+	resp, err := client.Do(req) //client.Get(targetUrl)
+	if err != nil {
+		log.Printf("Failure to do x509 req")
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		log.Printf("got error from call %s, url='%s'\n", resp.Status, url)
+		return nil, err
+	}
+	return ioutil.ReadAll(resp.Body)
 
-		resp, err := client.Do(req) //client.Get(targetUrl)
+}
+
+func getCertsFromServer(signer crypto.Signer, userName string, password []byte, baseUrl string, tlsConfig *tls.Config) (sshCert []byte, x509Cert []byte, err error) {
+	//First Do Login
+
+	tr := &http.Transport{
```

Comment:
Can you pick a longer, more descriptive variable name? It's not obvious what "tr" means.